From 346453faf91b19d228d25adc18f5d4e1c7c47f67 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Fri, 23 Aug 2024 03:13:30 -0500
Subject: [PATCH] fix: Fallback to readlink if macOS missing greadlink

* The 'g' prefixed coreutils on macOS are installed through
  'brew install coreutils'. While most macOS systems are using homebrew,
  there are situations, like in conda-forge builds, where the use of
  homebrew isn't supported. As GNU readlink and homebrew's greadlink
  have the same API, fall back to using readlink if greadlink isn't
  on PATH.
---
 HepMC3-config.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/HepMC3-config.in b/HepMC3-config.in
index 94fc1b4d..b395b6e8 100755
--- a/HepMC3-config.in
+++ b/HepMC3-config.in
@@ -5,7 +5,12 @@
 ## @configure_input@
 #The line below assures at least some relocation.
 if [ "$(uname)" == "Darwin" ]; then
-PACKAGE_BIN_DIR=$(dirname $(greadlink -f $0))
+    # greadlink is only available through 'brew install coreutils'
+    if [ ! -z "$(command -v greadlink)" ]; then
+        PACKAGE_BIN_DIR=$(dirname $(greadlink -f $0))
+    else
+        PACKAGE_BIN_DIR=$(dirname $(readlink -f $0))
+    fi
 else
 PACKAGE_BIN_DIR=$(dirname $(readlink -f $0))
 fi
-- 
2.46.0

